### PR TITLE
Add monitoring features to observe mailbox sizes

### PR DESCRIPTION
**Pull request by jbartelh**
In the case that an actor receives more messages than it processes, unbounded mailboxes can cause memory issues.
To spot the error-prone actor in such a case more easily, this PR replaces the default unbounded mailbox from Akka by an unbounded mailbox with additional monitoring capabilities. This mailbox keeps track of it's size, with a constant time complexity, which is used to:
- generate a log entry as soon as a configured mailbox-size-threshold is reached
- and a Prometheus metric ('actor_mailbox_size'), to track for each actor-class the number of messages inside their mailboxes.

### CODE REVIEW

**Review by Yannic92 — CHANGES_REQUESTED** [review]

File: services/utils/akka/src/main/java/org/eclipse/ditto/services/utils/akka/mailbox/MonitoredUnboundedMailboxType.java [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+      * creates a normal message queue or an instrumented queue, which keeps track of its size.
+      *
+      * @param owner the owner actor of this mailbox.
+      * @param system the actor system this mailbox is part of.
+      * @return a queue
+      */
+     @Override
+     public MessageQueue create(final Option<ActorRef> owner, final Option<ActorSystem> system) {
+         if (owner.isEmpty() || system.isEmpty()) {
+             throw new IllegalArgumentException("no mailbox owner or system given");
```
[the remark this answers is not in the document]
Comment (jbartelh) [reply services/utils/akka/src/main/java/org/eclipse/ditto/services/utils/akka/mailbox/MonitoredUnboundedMailboxType.java:86]: The actor will crash with an `ActorInitializationException`

File: services/utils/akka/src/main/java/org/eclipse/ditto/services/utils/akka/mailbox/MonitoredUnboundedMailboxType.java [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+         }
+
+         final ActorRef mailboxOwner = owner.get();
+
+         final Config mailboxConfig = config.getConfig(CONFIG_OBJECT_PATH);
+         final Pattern includeRegexFilters = Pattern.compile(mailboxConfig.getString(ACTORS_INCLUDE_REGEX_PATH));
+         final Pattern excludeRegexFilters = Pattern.compile(mailboxConfig.getString(ACTORS_EXCLUDE_REGEX_PATH));
```
[the remark this answers is not in the document]
Comment (jbartelh) [reply services/utils/akka/src/main/java/org/eclipse/ditto/services/utils/akka/mailbox/MonitoredUnboundedMailboxType.java]: Ah nice, done!

File: services/utils/akka/src/main/java/org/eclipse/ditto/services/utils/akka/mailbox/MonitoredUnboundedMailboxType.java [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+     /**
+      * Decides whether mailbox size of the actor shall be tracked or not.
+      *
+      * @param path the path of the actor.
+      * @param includeRegexFilters include filters, which determines that an actor shall be tracked.
+      * @param excludeRegexFilters exclude filters, which determines that an actor shall not be tracked (stronger than
+      * include filters).
+      * @return the made decision.
+      */
+     private static boolean trackActor(final ActorPath path, final Pattern includeRegexFilters,
```
[the remark this answers is not in the document]
Comment (jbartelh) [reply services/utils/akka/src/main/java/org/eclipse/ditto/services/utils/akka/mailbox/MonitoredUnboundedMailboxType.java]: Done!

File: services/utils/akka/src/main/java/org/eclipse/ditto/services/utils/akka/mailbox/MonitoredUnboundedMailboxType.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Yannic92) [reply services/utils/akka/src/main/java/org/eclipse/ditto/services/utils/akka/mailbox/MonitoredUnboundedMailboxType.java:86]: Do you think this is a reasonable behaviour for an actor? I would expect it to be restarted and crashing with the same reason over an over again.

**Review by Yannic92 — APPROVED** [review]

**Pull request merged**